StreamChatClient_v3/WebSocketClient/WebSocketClient.swift

```diff
@@ -212,18 +241,22 @@ extension WebSocketClient: WebSocketEngineDelegate {
             let messageData = Data(message.utf8)
             let event = try eventDecoder.decode(data: messageData)
             
-            if let event = event as? HealthCheck {
-                if connectionState.isConnected == false {
-                    connectionState = .connected(connectionId: event.connectionId)
-                    reconnectionStrategy.sucessfullyConnected()
-                }
-            }
-            
             middlewares.process(event: event) { [weak self] event in
                 guard let self = self, let event = event else { return }
-                self.notificationCenter.post(Notification(newEventReceived: event, sender: self))
+                
+                if let event = event as? HealthCheck {
+                    if self.connectionState.isConnected == false {
+                        self.connectionState = .connected(connectionId: event.connectionId)
+                        self.reconnectionStrategy.sucessfullyConnected()
```

Comment:
You can do that in `didSet` for `connectionState`

Reply:
What do you think are the benefits of doing it there instead of doing it here?

Reply:
I think it will be solid, because changing connection state should sometimes do a side effect and `didSet` is a best place for that. It was designed for that. We can be sure if the code will changed in future the behaviour will be the same. I think we can avoid mistakes.

Reply:
Ok, fair enough 👍

Reply:
fixed in 42bc4ba25099008d0f338e13d3bc4b769a6688d1